### DISCUSSION TITLE
Best practices for wifi and for elevated permission scripts

### DISCUSSION THREAD

**Discussion by Efrank700** [silitics/rugix#36] [Q&A] [answered]
What is the best way to provision wifi pre-emptively on devices? In general, i would use an nmcli script, but that requires elevated permissions that it seems `install` steps don't have.

**Comment by koehlma** [comment, marked as answer]
Install scripts should have full root permissions, but `nmcli` may need a running Network Manager instance, which does not exist while building the image. You also would not want it to, as the network devices of the build environment are different and should not be managed by the image being built.

The best solution to setup Wi-Fi may depend on your exact requirements. If you want/need to use Network Manager, you can have a look at the following recipe which is part of the thin-edge.io integration: https://github.com/thin-edge/tedge-rugpi-core/tree/v0.7/recipes/set-wifi. For our own deployments, we typically use Systemd Networkd together with WPA Supplicant, which can both be configured via files placed in the image (there are some online resources explaining how to configure this combination, e.g., https://wiki.somlabs.com/index.php/Connecting_to_WiFi_network_using_systemd_and_wpa-supplicant). Note that neither the Raspberry Pi OS nor the Debian variants of Rugpi come with Systemd Networkd preinstalled and, in case of Raspberry Pi OS, it is necessary to first uninstall Network Manager.

**Reply by Efrank700** [reply]
Solution actually turned out to be simpler than this. Install scripts have full permissions, but I didn't realize that I need to manually `chmod` the new recipe steps that get added in order to make them run in the image generation. 

No actual issue with the wifi setup. It turns out that was just the first custom step that ended up running based on my recipe arrangement.
